### DISCUSSION TITLE
Is there a "dart-only" cloud_firestore implementation?

### DISCUSSION THREAD

**Discussion by BenVercammen** [firebase/flutterfire#4550] [General] [answered]
I'm trying to build an application that's written in dart, without pulling in any of the flutter dependencies. However, I'd also like to use a firestore database for persistence.
Easliy enough I stumbled across the "cloud_firestore" package, but unfortunately it depends on flutter, which is something I don't really want, for a couple of reasons:
 - I can no longer run my example application as a dart program (I can run it as a flutter application, but I'm trying to avoid that)
 - I can no longer use reflection (mirrors) as flutter won't allow it (I can still use reflectable, but its reflection capabilities are limited)
 - I'd just like to use the application as a REST service, for which I have no reason to use flutter, AFAIK...

So my question is: why is the "cloud_firestore" package depending on flutter? Is there an alternative package I might try?

The only dependency I could find (at a quick glance) was for registering the FirebaseFirestoreWeb as a plugin. (Please not I'm new to dart and flutter, so I might get some terminology wrong...)
I'm aware that the actual repository is called "flutterfire" and is meant to bring firebase access in flutter, but that leaves me wondering why there's no "lower level dartbase" package?

**Comment by BenVercammen** [comment, marked as answer]
Okay, I had to look a bit further for a dart-only alternative: https://github.com/FirebaseExtended/firebase-dart

(https://stackoverflow.com/questions/57155885/is-there-a-dart-interface-to-firestore-with-api-as-in-https-firebase-google/57155947)

**Reply by Katekko** [reply]
it was discontinued
